commands/scan/scan.go

```diff
@@ -403,49 +385,58 @@ func (scanCmd *ScanCommand) createIndexerHandlerFunc(file *spec.File, entitledFo
 				}
 				graphScanResults, err := scangraph.RunScanGraphAndGetResults(scanGraphParams, xrayManager)
 				if err != nil {
-					log.Error(fmt.Sprintf("scanning '%s' failed with error: %s", graph.Id, err.Error()))
-					indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
+					log.Error(fmt.Sprintf("%s sca scanning '%s' failed with error: %s", clientutils.GetLogMsgPrefix(scanThreadId, false), graph.Id, err.Error()))
+					targetResults.AddError(err)
 					return
+				} else {
+					targetResults.NewScaScanResults(*graphScanResults)
+					targetResults.Technology = techutils.Technology(graphScanResults.ScannedPackageType)
 				}
-
-				scanResults := utils.Results{
-					ScaResults:          []*utils.ScaScanResult{{XrayResults: []services.ScanResponse{*graphScanResults}}},
-					ExtendedScanResults: &utils.ExtendedScanResults{},
+				if !cmdResults.EntitledForJas {
+					return
 				}
-				if entitledForJas && scanCmd.commandSupportsJAS {
```

Comment:
is there any reason why scan command won't be able to support JAS? since this condition is no longer checked..

Reply:
JAS is only supported for Docker images right now, added a comment